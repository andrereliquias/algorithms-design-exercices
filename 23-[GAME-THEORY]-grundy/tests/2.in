1000
0
1194
1321
4230
1959
6036
1622
9422
2279
2244
337
9138
5356
7509
5986
8980
9460
9191
8321
4335
5470
1191
3677
2806
9448
549
5210
6816
9882
4764
7097
2819
7037
9497
7049
8996
5532
9751
8418
8890
1994
8755
8028
7350
7342
5092
7409
6801
5361
6808
2215
831
9079
5892
4716
8526
7521
1004
6420
7402
6847
3517
220
3884
3013
8349
3958
9625
8099
2375
8514
1171
2208
7620
9600
629
2712
8087
7430
9152
5974
723
9983
5052
6616
5778
4656
5215
6782
1076
2616
3629
5672
3916
8592
9764
3343
2549
9388
1441
6004
8981
3691
9291
6600
3290
9920
390
1377
8429
9543
7351
9152
603
3481
6846
6381
9216
2060
4242
291
5756
8950
7042
750
7541
7885
4093
1168
7272
6613
7172
6252
303
7542
3931
4672
7461
5400
7128
6968
4942
4478
7199
6625
9038
5123
3005
8254
7184
7247
9624
4018
6196
6666
5847
4815
4550
9940
5984
2900
7631
4234
231
9013
1775
4162
3684
315
640
812
7283
5583
6369
5560
2207
5407
683
6291
3660
8946
4617
4362
4042
1891
1027
9889
6707
6656
9828
3769
635
8537
8003
866
7549
857
6107
1232
1172
6747
2044
9534
3408
8414
5094
6694
4899
6856
2985
9638
5801
7602
3999
9843
571
6106
9731
7278
2761
636
1046
3396
251
128
5341
7800
985
2526
110
2157
352
3234
2769
3760
2726
8942
1533
7625
5797
4518
7262
2676
3198
1260
3597
3769
7366
4406
2126
1206
6121
3172
5681
6372
4379
2101
5250
5364
4627
5361
8600
4979
8595
1369
9819
1320
1389
1351
23
8266
6948
7285
941
1224
9624
5618
4993
8069
23
8198
9275
6145
2449
6034
3595
6828
8135
8846
2192
2762
5285
791
8820
3879
3239
9717
6278
5708
2146
7380
3973
9094
4664
4914
317
5366
1610
6390
3434
1634
4587
3787
8858
7036
9822
2452
4943
7956
2376
7135
1796
7661
9005
1695
2618
3323
1411
9975
9031
3558
7354
3003
3730
3096
8996
5127
8463
606
2595
2975
3319
8261
7842
2176
5297
7663
5707
1318
6697
9163
8453
8494
6823
8536
188
520
1858
2678
494
888
6236
8928
4970
1045
2023
3966
6172
1564
5651
8767
5619
8970
7027
3460
2224
3402
2201
7931
4720
8898
7093
4251
8470
4995
2787
9737
6594
4645
2415
8167
6613
9730
7094
1582
774
196
6627
6946
2839
2277
6791
8458
2325
3818
2996
4549
8299
6276
3559
3019
5174
1730
7270
4722
7804
56
4459
4397
5781
6874
2564
2393
6603
9658
5054
7378
9854
1681
5402
3772
5037
2193
3308
8442
7090
6305
4069
5388
2580
7628
8407
8833
437
6756
3555
8241
7891
8014
2638
4750
5966
5202
7143
2568
5938
3276
1024
6871
6036
7506
642
2151
9699
3950
592
6788
1333
5741
3254
3914
4447
2740
3825
4884
9496
7380
3125
8465
6472
6842
3215
3516
2043
1436
7164
9060
4712
8188
5930
1826
5693
6572
3978
6470
1601
5649
4336
2934
1389
7591
7927
5837
1409
1752
720
904
210
4924
447
7762
2844
3662
1277
5966
6178
8441
5026
889
7708
955
3795
4479
8607
7773
949
207
3421
6364
3141
5890
5033
2147
1726
6442
3899
3525
8425
5188
8450
8873
2949
1293
3613
5306
7260
9791
3746
3364
1759
2532
4319
5554
7012
2925
4405
9040
4211
8905
5403
8432
4794
1515
578
6520
7957
5556
1124
7461
743
9574
6333
4772
1945
9946
77
283
815
4902
3647
2574
8514
9046
8128
5525
3049
2532
4564
7261
2515
1045
5692
7310
3639
7349
4908
2675
2904
6032
135
4726
6684
7547
9498
9709
8571
653
9992
9386
6635
4718
1959
5148
4842
1165
1751
7891
4776
7394
6230
7292
9518
3000
5680
3157
348
587
6911
4331
6620
7046
9058
4382
5671
9634
4090
4241
1366
5161
4706
8001
9879
7744
4227
5799
9989
7057
3689
5843
4450
9920
3134
5046
2919
9893
9282
4347
480
6192
8678
8179
4316
8814
2560
9987
8448
7730
5307
9814
2890
1091
8893
3847
8835
4198
9646
8823
1254
4413
4666
6783
4332
8879
1828
8331
9851
2189
3756
330
8381
3512
9588
3776
2326
3226
3762
773
2034
147
1665
4924
1238
557
9850
1152
4755
9495
1053
7088
3908
6798
3870
9319
6756
6777
8728
6606
8966
2483
8015
8426
5996
8681
2201
9401
1906
7042
1252
5020
8269
2917
9944
585
4553
9794
1737
386
366
2791
7474
5353
667
2422
4672
7423
9199
4478
4028
9244
6962
3121
7669
4036
1801
948
3436
4787
9069
4688
9807
7337
8684
828
7923
3236
621
9660
3622
2067
3529
1095
7420
5276
4596
3170
2698
3794
7649
7806
4116
5689
2005
2863
803
3807
3811
5318
9673
3959
1084
9479
1295
9768
306
9218
4082
2007
9957
7704
5153
4564
9878
2572
9840
4473
6822
3617
9346
5549
1422
4541
2316
3427
7404
3119
8313
2294
9516
7985
6253
599
8542
7548
1445
8849
7845
6606
1934
7801
5388
7087
3443
6344
737
3283
1895
7559
6900
1240
4186
9401
6860
6502
3906
4264
699
3298
6558
214
1282
3889
1892
9825
2515
4416
9752
359
1021
1685
9239
6409
9851
2682
2752
587
7044
4647
8147
3943
6967
3411
4422
3826
9914
9407
9169
612
2704
6805
1906
3987
693
3798
4890
3209
8215
5720
4647
314
7405
3886
6724
7255
7647
554
7842
4690
6281
7067
9712
4326
478
4133
9231
1470
3539
8400
2082
7323
6283
3988
1309
6977
8866
7278
1264
7080
2997
5911
8473
1480
875
5196
8735
8522
6830
7655
3211
4189
5801
2922
8515
7358
8134
7745
9907
2752
7223
3067
74
3506
8135
2462
1561
7000
818
3904
5158
3815
9815
3630
5295
690
9906
5108
290
6735
3841
3502
923
720
7503
516
9157
6716
9340
9063
9468
6563
2130
620
1147
264
3082
3787
8343
4979
7691
3500
8794
7505
8209
5167
9274
8114
1353
643
4848
6273
5224
6850
6994
2726
7366
7229
520
6706
6292
1066
4347
