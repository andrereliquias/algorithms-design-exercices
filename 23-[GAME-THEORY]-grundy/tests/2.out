Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Teco
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Teco
Teco
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Teco
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Teco
Teco
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Teco
Teco
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Teco
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Teco
Teco
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Teco
Teco
Teco
Tico
Teco
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Teco
Tico
Teco
Tico
Tico
Teco
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Teco
Tico
Tico
Tico
Teco
Tico
Tico
Teco
Tico
Tico
Tico
Tico
Teco
Tico
Tico
Teco
